### ISSUE TITLE
Add a resume parser

### ISSUE THREAD

**Issue by Vedantjn** [Vedantjn/easy-apply#1]
Add a resume parser that parses the resumes that are being uploaded, then fill them in a form that asks for only the essential information, and makes the job applying process an easy task as compared to Workday or any other platform.

**Comment by Ayush05m** [comment]
Hy, I will do the task because I want to explore how this can be done and implemented, we should approach with the cloud based services for better efficiency. Have you made the backend of the project, without backend we have to go with the react libraries which may not be as efficient as the cloud based.

**Comment by Vedantjn** [comment]
Hey @Ayush05m, thanks for your input!

Could you clarify which specific cloud-based service you're referring to for the resume parsing? I'd also love to know which particular React libraries you had in mind for implementing this feature if we were to go the frontend-only route. For now, I'd prefer to explore free solutions. Regarding the backend, I'd encourage you to go through the codebase first to get a better understanding of the current setup, and we can plan its implementation based on that.

**Comment by yashbudhia** [comment]
is it fine if i use LLM's for it

**Comment by Vedantjn** [comment]
@yashbudhia exactly how?

**Comment by yashbudhia** [comment]
Once The user uploads resume via a post method , the method will then be passed onto pdfminer to covert it into text which then will be passed through structured prompts to groq chat client to extract important details like -
Name
Email
Phone Number
Education
Work Experience
Skills
Certifications
LinkedIn/GitHub Profile Links

, How's this for a start?

**Comment by Vedantjn** [comment]
@yashbudhia that sounds amazing, I'm assigning this to you, also go through the whole codebase first.

**Comment by yashbudhia** [comment]
yup groq is free but has token limits , im assuming for starting its fine , as the website grows , we will just upgrade , also i will require some time to implement this . Thanks for assigning this to me

**Comment by Ayush05m** [comment]
I'm sorry for the late reply but that's a great idea and interesting to implement. Although I'll search for alternate options as well.

**Comment by Vedantjn** [comment]
@Ayush05m, for now, I've assigned this to @yashbudhia, you can checkout different issues.
